loadtest/control/simulcontroller/actions.go

```
@@ -33,3 +39,108 @@ func (c *SimulController) reload(full bool) control.UserActionResponse {
+	// TODO: change this to make sure we don't pick the same team we are
+	// currently in.
+	team, err := u.Store().RandomTeamJoined()
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	if err := u.SetCurrentTeam(&team); err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	if _, err := u.GetChannelsForTeamForUser(team.Id, u.Store().Id()); err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	if err := u.GetChannelMembersForUser(u.Store().Id(), team.Id); err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	if _, err := u.GetTeamsUnread(""); err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	// TODO: use more realistic data.
+	var userIds []string
+	userIds = append(userIds, u.Store().Id())
+	if err := u.GetUsersStatusesByIds(userIds); err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	c.status <- c.newInfoStatus(fmt.Sprintf("switched to team %s", team.Id))
+
+	// We should probably keep track of the last channel viewed in the team but
+	// for now we can simplify and randomly pick one each time.
+
+	return switchChannel(u)
+}
+
+func switchChannel(u user.User) control.UserActionResponse {
+	team, err := u.Store().CurrentTeam()
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	} else if team == nil {
+		return control.UserActionResponse{Err: control.NewUserError(fmt.Errorf("current team should be set"))}
+	}
+
+	// TODO: change this to make sure we don't pick the same channel we are
+	// currently in.
+	channel, err := u.Store().RandomChannelJoined(team.Id)
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	if current, err := u.Store().CurrentChannel(); err != nil {
+	} else if current != nil {
```

Just saw this. Need to return the error.